### PR TITLE
fix: preserve edited Codex user configuration during reinstall and repair

### PR DESCRIPTION
**Pull request by shoyann**
## What Changed

After ECC creates `~/.codex/config.toml` or `~/.codex/AGENTS.md`, reinstall and repair can overwrite subsequent user edits. Preserve these shared files when their contents differ from the last installed digest, or when an older ledger has no digest to establish ownership of the current bytes.

- Keep user content byte-for-byte, warn that future configuration updates require manual application, and remove preserved files from the managed ledger so uninstall leaves them alone.
- Apply the user-owned-file guard during Codex repair, including files present before the first install and files retained across selective reinstalls.
- Recheck before writes. Keep previous digests for shared configurations that repair did not write, preventing state refresh from claiming edits made after inspection.
- Continue updating unchanged scaffolds and restoring missing ones.

## Why This Change

Addresses finding 6 in #3116. This is whole-file preservation, without TOML/Markdown merging or new dependencies. It addresses one finding of the multi-part issue, so this PR does not close the entire issue. Other managed files retain their existing update behavior.

## Testing Done

- [x] Manual testing completed: native Windows CLI install → edit both files → repair preview → repair twice → uninstall; personal bytes remain identical.
- [x] 28 new automated regression tests pass, covering reinstall, repair, dry run, selective reinstall, repeated repair, old ledgers, preexisting configurations, unchanged/missing scaffolds, concurrent edits, and uninstall.
- [x] Five existing related suites pass: ownership guard, lifecycle, executor, selective reinstall, and Claude skill migration.
- [x] Edge cases considered and tested. Initial preservation regression tests failed before the fix; disabling the repair digest protection reproduces all four new repair-race failures.
- [x] ESLint and `git diff --check` pass.
- [ ] Full repository suite (`node tests/run-all.js`) — not run.

Validated on Windows with Node v24.15.0. Existing suites skip some OS-specific checks; no Linux/macOS run is claimed. Scoped coverage for `codex-user-config.js` and `ownership-guard.js`: 97.37% lines, 85.71% branches, 100% functions; coverage thresholds pass. This is not whole-repository coverage.

## Type of Change

- [x] `fix:` Bug fix

## Security & Quality Checklist

- [x] No secrets or API keys committed (specified token patterns checked).
- [x] No sensitive data exposed in committed tests or output.
- [x] Follows conventional commits format.
- [x] Path containment and no-follow reads protect the configuration digest check; independent code/security review completed.

No JSON, shell scripts, dependencies, or package metadata changed. The new internal helper is covered by the existing `scripts/lib/` publish surface. No new skill, command, agent, hook, or CLI entry point is introduced.

## Documentation

- [x] Added comments for ownership and digest preservation logic.
- [x] Preservation warnings explain that the file becomes user-managed and future configuration updates must be applied manually.

### CODE REVIEW

File: tests/lib/install-codex-config-preservation.test.js [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        assert.equal(report.results[0].status, 'ok');
+      }
+      assertPreserved(fixture, name, content);
+      const refreshedOperation = readInstallState(fixture.statePath).operations.find(operation => (
+        operation.destinationPath === fixture.destination(name) && operation.ownership === 'managed'
+      ));
+      if (refreshedOperation) {
+        assert.equal(refreshedOperation.contentSha256, previousOperation.contentSha256,
+          'Repair must retain the previous digest for configuration it did not write');
+      }
```
[the remark this answers is not in the document]
Comment (shoyann) [reply tests/lib/install-codex-config-preservation.test.js]: Addressed in 08094a3. Added unconditional presence assertions at both ledger-check sites. The bridge/no-op repair tests now require the previous managed entry to exist before comparing its digest; the partial reinstall checkpoint tests do the same. A missing entry can no longer silently skip the digest assertion.

Validation: all 28 configuration-preservation tests pass on Windows; ESLint and `git diff --check` pass. This follow-up changes only test assertions and addresses this CodeRabbit comment. The separate Greptile concurrency finding is not addressed by this commit.
